packages/react-utilities/src/hooks/useId.ts

```diff
@@ -21,5 +21,11 @@ export function resetIdsForTests(): void {
 export function useId(prefix?: string, providedId?: string): string {
   const contextValue = useSSRContext();
 
-  return React.useMemo(() => providedId || `${prefix}${++contextValue.current}`, [prefix, providedId, contextValue]);
+  return React.useMemo(() => {
```

Comment:
Not the point of this PR but it's actually a little scary that this uses `useMemo`, at least with respect to how I originally intended it to behave when I wrote it in v8. My original intended behavior was that this should generate a unique ID when it's initially run and *never change* (though potentially making it respect updates to `providedId` could be reasonable). The React docs clearly state that `useMemo` is a performance optimization, not a guarantee that the calculation will only ever run once.

It looks like @layershifter changed this in #17924 for SSR compatibility, but I'm not clear on whether the `useMemo` change was strictly necessary for SSR support or not.

Reply:
Also, having the ID begin with a number is allowed in HTML 5 but wasn't in HTML 4, so I'm not sure if it has the potential to cause any problems due to outdated expectations in other libraries or something.

Reply:
We could potentially work around the HTML4 requirement if you add a predefined prefix like 'fui-' if no prefix was specified.

Reply:
To be clear I'm not sure if the HTML4 thing still matters, since HTML5 has been around quite awhile now. I just don't trust libraries/etc to be up to date, so it seems maybe safest to include some prefix anyway (also looks nicer IMO).

Reply:
`contextValue` is an object that can cause the id to recompute. Recomputing the Id incremenents a counter which will actually change the id.

The server context value and client context value are different, so without `useMemo` you would get hydration errors (or worse?) because the id in the client is different from the server render.

Reply:
I like the idea of providing a default "fui-" prefix. What do you think: add it to this PR or make a follow up to add it?

Reply:
is HTML4 still a requirement for us ? even IE11 supports HTML5

Reply:
I feel that generating IDs like `fui-123` reads nicer than plain `123` and makes it clearer where an ID came from. Granted these are all autogenerated and being "nice to read" isn't really a requirement but it fits with our practice of prefixing classnames. I doubt HTML4 is still a requirement (though I don't know for sure) but it doesn't seem like a bad thing to support in this case.

Reply:
> I feel that generating IDs like fui-123 reads nicer than plain 123 and makes it clearer where an ID came from.

Agreed, IMO "reads nicer" is a good enough argument to do it.

> is HTML4 still a requirement for us ? even IE11 supports HTML5

Probably not, like I said before I'm just suspicious of *something, somehow* getting tripped up by it due to being super outdated? But even that is relatively unlikely at this point.

Reply:
I see that the change to `fui` prefix broke no snapshot tests, which means that either not all components are tested correctly or not components are actually using `useId` without a prefix.

Could we instead make the `prefix` a required arg ?

Reply:
Well, it won't affect any v8 components so that eliminates much potential snapshot breakage.

Outside of v8 (`packages/react`) I see the following cases where `useId()` is not provided a prefix:
- Coachmark.stories.tsx -- vr-test with no snapshots (it has a Screener snapshot but AFAIK that's an image so I wouldn't expect it to fail with this change). Also this is using the v8 version of the hook.
- SelectedPerson.tsx in react-experiments -- imports the `useId()` implementation from `@fluentui/react-hooks`. This ultimately calls [`getId()`](https://github.com/microsoft/fluentui/blob/master/packages/utilities/src/getId.ts#L6) which does provide a default prefix.
- react-select -- uses it in Storybook stories so I wouldn't expect snapshot breakage.
- react-slider -- uses it in Storybook stories so I wouldn't expect snapshot breakage.
- useId() tests -- the tests in this package test this functionality. No breakage.

Based on this it looks like no v9 components are using the hook without providing a prefix.

`useId()` has already been released as a stable API in RC and making `prefix` required is a breaking change and I don't think it meets our bar for something we'd change. That said it's a pretty simple fix for anyone is does break.

My feeling is we should keep `prefix` optional with a default value as this aligns the v9 behavior of this hook with the v8 implementation and it won't break any users.